### DISCUSSION TITLE
Thumbnail Generation Settings

### DISCUSSION THREAD

**Discussion by Xipherisroot** [owncast/owncast#1885] [Q&A]
I get some bad CPU use spikes from ffmpeg seemingly when its generating a thumbnail for the stream, I'm using passthrough so wouldn't be related to transcoding.

Where can I set the interval for generation, set a static thumbnail or otherwise disable thumbnail generation? I'm struggling to find where this is defined/triggered.

**Comment by gabek** [comment]
This isn't something that's customizable. You can alter the source, however.  https://github.com/owncast/owncast/blob/develop/core/transcoder/thumbnailGenerator.go#L31

**Reply by Xipherisroot** [reply]
That is a bummer...

I did pull 0.0.12 and changed it which resolved the issue in a big way, set it to every 24hr, but even before I made the change I couldn't get 0.0.12's current dev build to work fully, web interface failed to load, RTMP stream worked great though! ultimately defeated the purpose for me not having the end user interface working correctly, bummer again...

Ultimately I went back to 0.0.11 but did a source build of FFMPEG 5.0.1 and its gone from 2-3 seconds of 100% CPU every 20 seconds to barely a split second blip of 30-40%, load average is back down to pretty much nothing, that's resolved enough for me at the moment!

Any chance that this will be a configurable variable in the future? The sort of stuff I stream hardly needs a new thumb generated ever beyond the initial one, I figure there's a lot of content that doesn't really need a preview picture generated that often.
